src/pip/_internal/utils/setuptools_build.py

```diff
@@ -1,13 +1,28 @@
+import sys
+
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from typing import Optional
 # Shim to wrap setup.py invocation with setuptools
 #
 # We set sys.argv[0] to the path to the underlying setup.py file so
 # setuptools / distutils don't take the path to the setup.py to be "-c" when
 # invoking via the shim.  This avoids e.g. the following manifest_maker
 # warning: "warning: manifest_maker: standard file '-c' not found".
-SETUPTOOLS_SHIM = (
+_SETUPTOOLS_SHIM = (
     "import sys, setuptools, tokenize; sys.argv[0] = {0!r}; __file__={0!r};"
     "f=getattr(tokenize, 'open', open)(__file__);"
     "code=f.read().replace('\\r\\n', '\\n');"
     "f.close();"
     "exec(compile(code, __file__, 'exec'))"
 )
+
+
+def make_setuptools_shim_args(setup_py_path, unbuffered_output=False):
+    # type: (str, Optional[bool]) -> list
```

Comment:
I believe this can be `bool` instead of `Optional[bool]`. Also, the return value can be `List[str]`. I would also document `unbuffered_output` with `:param unbuffered_output:` in a docstring.